src/Stratis.Bitcoin.Features.Consensus/PowConsensusValidator.cs

```diff
@@ -492,37 +492,22 @@ private long CountWitnessSignatureOperation(Script scriptSig, Script scriptPubKe
                 return 0;
 
             WitProgramParameters witParams = PayToWitTemplate.Instance.ExtractScriptPubKeyParameters2(scriptPubKey);
-            if (witParams != null)
-                return this.WitnessSigOps(witParams, witness, flags);
-
-            if (scriptPubKey.IsPayToScriptHash && scriptSig.IsPushOnly)
-            {
-                byte[] data = scriptSig.ToOps().Select(o => o.PushData).LastOrDefault() ?? new byte[0];
-                Script subScript = Script.FromBytesUnsafe(data);
 
+            if (witParams == null && scriptPubKey.IsPayToScriptHash && scriptSig.IsPushOnly)
```

Comment:
This two lines done make sense so much
```
WitProgramParameters witParams = PayToWitTemplate.Instance.ExtractScriptPubKeyParameters2(scriptPubKey);
 if (witParams == null && scriptPubKey.IsPayToScriptHash && scriptSig.IsPushOnly)
        witParams = PayToWitTemplate.Instance.ExtractScriptPubKeyParameters2(scriptPubKey);
```

If in the first call witParams is null then it will also be null in the second case right?

They are the same call
```
WitProgramParameters witParams = PayToWitTemplate.Instance.ExtractScriptPubKeyParameters2(scriptPubKey);
                     witParams = PayToWitTemplate.Instance.ExtractScriptPubKeyParameters2(scriptPubKey);

```